#! /usr/bin/env python

PACKAGE='aricc_2d_vision'
try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("debug", bool_t, 0, "debug", False)
gen.add("invert_color", bool_t, 0, "invert_color", False)
gen.add("use_P", bool_t, 0, "use camera parameters", False)
gen.add("board_type", str_t, 0, "board type", "chess")
gen.add("columns", int_t, 0, "columns", 1, 0, 10)
gen.add("rows", int_t, 0, "rows", 1, 0, 10)
gen.add("max_board", int_t, 0, "rows", 1, 0, 10)
gen.add("border_thresh", int_t, 0, "border threshold", 30, 0, 100)
gen.add("rect_size_cols", double_t, 0, "size of columns", 0.01, 0, 1.0)
gen.add("rect_size_rows", double_t, 0, "size of rows", 0.01, 0, 1.0)

exit(gen.generate(PACKAGE, "aricc_2d_vision", "CheckBoardDetector"))
